### PR TITLE
Fix java.sql.Date serialize error by convert it to java.util.Date

### PR DESCRIPTION
**Pull request by JueLance**
## Purpose of the pull request

please see this issue: https://github.com/apache/dolphinscheduler/issues/12983

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/json/DateJsonSerializer.java [hunk 1]
```diff
@@ -33,6 +34,14 @@ public class DateJsonSerializer extends JsonSerializer<Date> {
 
     @Override
     public void serialize(Date value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
-        gen.writeString(DateUtils.dateToString(value));
+        Date temp = value;
+        // In the internal implementation of JDK, sql.Date is inherited from util.Date, and sql.Date has rewritten many methods in util.Date, but it has not been implemented.
```
Comment (fuchanghai) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/json/DateJsonSerializer.java]: plz write the full path to the class ,like ``` java.util.Date```

**Comment by fuchanghai** [comment]
@SbloodyS plz help to run CI

**Comment by SbloodyS** [comment]
> @SbloodyS plz help to run CI

Done.

**Comment by JueLance** [comment]
@SbloodyS Checks  run with 2 failing result. Do we need care about it?

**Comment by SbloodyS** [comment]
> @SbloodyS Checks run with 2 failing result. Do we need care about it?

Yes. All marked with the `Required` tag must pass.

**Comment by JueLance** [comment]
> > @SbloodyS Checks run with 2 failing result. Do we need care about it?
> 
> Yes. All marked with the `Required` tag must pass.

How do I can correct them? Please help me, actually I didn't touch e2e module, I don't know why these modules was run failed.

**Comment by fuchanghai** [comment]
plz，check the error log



---Original---
From: ***@***.***&gt;
Date: Fri, Dec 2, 2022 10:50 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [apache/dolphinscheduler] fixed java.sql.Date serialize error byconvert it to java.util.Date (PR #13050)




   
@SbloodyS Checks run with 2 failing result. Do we need care about it?
  
Yes. All marked with the Required tag must pass.
  
How do I can correct them? Please help me, actually I didn't touch e2e module, I don't know why these modules was run failed.

**Comment by fuchanghai** [comment]
@zhuangchong @SbloodyS plz help to run CI ,when you receive and have free time

**Comment by JueLance** [comment]
@fuchanghai @SbloodyS @zhongjiajie 现在需要我怎么做才能合并PR?

**Comment by JueLance** [comment]
> @fuchanghai @SbloodyS @zhongjiajie 现在需要我怎么做才能合并PR?

@caishunfeng

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/json/DateJsonSerializer.java [hunk 2]
```diff
@@ -33,6 +34,15 @@ public class DateJsonSerializer extends JsonSerializer<Date> {
...
-        gen.writeString(DateUtils.dateToString(value));
+        Date temp = value;
+        // In the internal implementation of JDK, java.sql.Date is inherited from java.util.Date, and java.sql.Date has
+        // rewritten many methods in java.util.Date, but it has not been implemented.
+        // Later, when formatting, you need to call the method in Date to report an error.
+        if (value instanceof java.sql.Date) {
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTime(value);
+            temp = calendar.getTime();
+        }
```
Comment (ruanwenjun) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/json/DateJsonSerializer.java]: The right way is to fix this at `DateUtils.dateToString`, rather than modify this class.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/DateUtils.java [hunk 3]
```diff
@@ -75,6 +75,15 @@ private static LocalDateTime date2LocalDateTime(Date date) {
...
+        // In the internal implementation of JDK, java.sql.Date is inherited from java.util.Date, and java.sql.Date has
+        // not been implemented toInstant() method. If call this method, it will throw new
+        // java.lang.UnsupportedOperationException(). Here just convert java.sql.Date to java.util.Date
+        // to avoid the error.
+        if (date instanceof java.sql.Date) {
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTime(date);
+            date = calendar.getTime();
+        }
         return LocalDateTime.ofInstant(date.toInstant(), zoneId);
```
Comment (ruanwenjun) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/DateUtils.java]:
```suggestion
        return LocalDateTime.ofInstant(Instant.of(date.getTime()), zoneId);
```

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/json/DateJsonSerializer.java [hunk 4]
```diff
@@ -17,17 +17,15 @@
 
 package org.apache.dolphinscheduler.service.json;
 
+import com.fasterxml.jackson.core.JsonGenerator;
```
Comment (ruanwenjun) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/json/DateJsonSerializer.java]: Please revert this change.

**Review by ruanwenjun — DISMISSED** [review]

**Review by ruanwenjun — DISMISSED** [review]

**Review by ruanwenjun — APPROVED** [review]

**Comment by ruanwenjun** [comment]
Merged, thanks for your PR.
